### PR TITLE
Separate C#-specific properties from general application properties.

### PR DESCRIPTION
**Pull request by melytc**
This is the first step to bringing the VB Application properties; separating the C#-specific properties would allow us to have one _base_ ApplicationPropertyPage rule file where the common properties live, and have .CSharp and .VisualBasic files for those who are specific to the language.

In the case of WPF and WinForms, my suggestion is that their properties live in the .VisualBasic file and have their visibility conditioned to the `UseWinForms`/`UseWPF` flags.

For reference:
image

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8151)

### CODE REVIEW

**Comment by melytc** [comment]
Just noticed that the properties fot tagged as new, even if I only moved their location. I'll see if I can keep their previous value so we don't have to do another localization pass.

**Comment by tmeschter** [comment]
If there is useful WinForms/WPF functionality in the VB Application page we should consider bringing it to C#, as well. Historically lots of aspects of C# and VB were handled by different teams, so the differences may be accidental rather than intentional.

That being said, I wouldn't hold up the change for that.

**Comment by tmeschter** [comment]
We have a unit test that verifies some of our VSIX contents. If you run the test from the command line (e.g. by running `build.cmd`) I think it will even put the commands to update the baseline in the clipboard.

**Comment by melytc** [comment]
> If there is useful WinForms/WPF functionality in the VB Application page we should consider bringing it to C#, as well. Historically lots of aspects of C# and VB were handled by different teams, so the differences may be accidental rather than intentional.

That's interesting. I was separating properties based on the old property pages; I'll add those properties in the next PR around WPF/WinForms, keeping this in mind :)

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ApplicationPropertyPage.CSharp.xaml [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  <Rule.Categories>
+    <Category Name="Resources"
+              DisplayName="Win32 Resources"
+              Description="Resource settings for the application." />
+    <Category Name="Packaging"
+              DisplayName="Packaging"
+              Description="Packaging settings for the application." />
+  </Rule.Categories>
+
+  <BoolProperty Name="TargetMultipleFrameworks"
```
[the remark this answers is not in the document]
Comment (melytc) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ApplicationPropertyPage.CSharp.xaml:16]: TargetFramework properties make sense to bring to VB projects, too; with the exception maybe of `TargetMultipleFrameworks` in WinForms projects?

**Review by tmeschter — APPROVED** [review]

**Pull request merged**